quill-core/src/main/scala/io/getquill/norm/ApplyMap.scala

```diff
@@ -100,6 +106,15 @@ object ApplyMap {
         val t2 = BetaReduction(d, c -> Property(t, "_2"))
         Some(Map(Join(tpe, a, b, iA, c, onr), t, Tuple(List(t1, t2))))
 
+      // a.*join(b.map(c => d).distinct).on((iA, iB) => on)
+      //    a.*join(b.distinct).on((iA, c) => on[iB := d]).map(t => (t._1, d[c := t._2]))
```

Comment:
This transformation doesn't seem valid. In `b.map(c => d).distinct`, it'll return distinct mapped values, in `b.distinct` it'll return distinct records considering all columns

Reply:
Good point. I took this out.